v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/schema/SourceTableSchema.java

```diff
@@ -0,0 +1,133 @@
+ * added here as and when required.
+ */
+@AutoValue
+public abstract class SourceTableSchema implements Serializable {
+  private static final String AVRO_SCHEMA_RECORD_DEFAULT_NAME = "BulkReaderRecord";
+  public static final String PAYLOAD_FIELD_NAME = "payload";
+  public static final String READ_TIME_STAMP_FIELD_NAME = "read_timestamp";
+
+  public abstract String tableSchemaUUID();
+
+  public abstract String tableName();
+
+  // Source Schema from metadata tables. SourceColumnType is similar to
+  // com.google.cloud.teleport.v2.spanner.migrations.schema
+  /* TODO(vardhanvthigle):
+   * Simplify the builder by integrating with common type mapping converters.
+   * The consumer of the builder does not need to build avro schemas once there's a defined mapping between `SourceColumnType` and Avro Schema.
+   * The todo is completely internal to reader and does not in any way change the contract that reader exposes to outside.
+   */
+  public abstract ImmutableMap<String, SourceColumnType> sourceColumnNameToSourceColumnType();
+
+  // Mapped Avro Schema (to unified types) that each row will carry.
+  public abstract Schema avroSchema();
+
+  /*
+   * TODO(vardhanvthigle): This will be not needed once we have complete type mapping and builder is simplified.
+   * This todo is internal to the reader and does not change the contract.
+   */
+  public static FieldAssembler<RecordDefault<Schema>> avroSchemaFieldAssembler() {
+    /* Note: DataStream `read_timestamp` is encoded as `timeMillis`. We use `timeMirco` since:
+     * 1. To achieve 1 GBPs performance (potentially on a single table), we will be writing 1 to 10 million records per second.
+     * (Assuming around 100 to 1000 bytes per record)
+     * 2. That level of throughput will make milliseconds less precise than what we need.
+     * 3. For example, for writing 1 million records per second, if we are batching 100 records, by Little's law,
+     *    we get a mean latency of around 100 micro seconds.
+     * 4. A precision finer than micro second granularity will be severely affected by clock skew across machines.
+     */
+    Schema timeMicroType = LogicalTypes.timeMicros().addToSchema(Schema.create(Schema.Type.LONG));
+    return org.apache.avro.SchemaBuilder.record(AVRO_SCHEMA_RECORD_DEFAULT_NAME)
+        .fields()
+        .name(READ_TIME_STAMP_FIELD_NAME)
+        .type(timeMicroType)
+        .noDefault()
+        .name(PAYLOAD_FIELD_NAME)
+        .type()
+        .record(PAYLOAD_FIELD_NAME)
+        .fields();
+  }
+
+  public Schema getAvroPayload() {
+    return avroSchema().getField(PAYLOAD_FIELD_NAME).schema();
+  }
+
+  /* TODO(vardhanvthigle@): Add more information as needed by other modules, like:
+   * primary keys, indexing information, foreign key constraints etc.
+   */
+
+  public static Builder builder() {
+    var builder = new AutoValue_SourceTableSchema.Builder();
+    builder.setTableSchemaUUID(UUID.randomUUID().toString());
```

Comment:
I also don't see the generation of `tableSchemaUUIDToSourceSchemaReference<>` map. Should'nt that be added in this PR as well?